pkg/monitortests/kubeapiserver/legacykubeapiservermonitortests/early_e2e_disruption.go

```diff
@@ -0,0 +1,53 @@
+package legacykubeapiservermonitortests
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+)
+
+// amount of time after E2E tests start that we consider "early"
+const earlyE2ECheckDuration = 3 * time.Minute
+
+func testEarlyE2EAPIServerDisruption(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
+	const testName = "record api server disruption that occurs early in E2E tests"
+
+	// regex to match the backends we care about
+	var earlyE2EBackendRe = regexp.MustCompile(`^(cache-)?(openshift|kube|oauth)-api-`)
+
+	var e2eStart time.Time
+	var tests []*junitapi.JUnitTestCase
+	for _, event := range events {
+		switch event.Source {
+		case monitorapi.SourceE2ETest:
+			if e2eStart.IsZero() {
+				e2eStart = event.From
+			}
+		case monitorapi.SourceDisruption:
+			if event.Message.Reason == monitorapi.DisruptionBeganEventReason &&
+				event.From.Sub(e2eStart) < earlyE2ECheckDuration &&
+				earlyE2EBackendRe.MatchString(event.Locator.Keys["backend-disruption-name"]) {
+				tests = []*junitapi.JUnitTestCase{
+					{
+						Name: testName,
+						FailureOutput: &junitapi.FailureOutput{
+							Output: fmt.Sprintf("found disruption %v after E2E tests started", event.From.Sub(e2eStart)),
```

Comment:
When this triggers, it will be helpful for the person looking at it to see if there were multiple, not just the first we found, so probably good to assembled the list of all disruption intervals we found and then pack them all into one FailureOutput message.